### ISSUE TITLE
Leader election failed when node was not ready

### ISSUE THREAD

**Issue by bluven** [operator-framework/operator-sdk#3462]
## Bug Report

**What did you do?**

I developed a operator named helm-controller.  I used deployment to run this operator, the deployment's replicas was set to 1. 

**What did you expect to see?**

New pod should get the leader lock when old leader pod could not work.                           

**What did you see instead? Under which circumstances?**

 Pod `helm-controller-5f89f89569-554ld`  kept trying to get leader lock, as the pod `helm-controller-5f89f89569-mkksl` was stucked on terminating status because the node was borken (I stopped docker and kubelet to achieve this goal)

```
kubectl get nodes
NAME           STATUS     ROLES    AGE    VERSION
bluven-k8s-1   Ready      master   153d   v1.17.3
bluven-k8s-2   Ready      <none>   153d   v1.17.3
bluven-k8s-3   Ready      <none>   153d   v1.17.3
bluven-k8s-4   NotReady   <none>   153d   v1.17.3
```

```
kubectl logs -n operator -f helm-controller-5f89f89569-554ld                                  

2020-07-20T06:18:32.342Z	INFO	leader	Not the leader. Waiting.
2020-07-20T06:18:49.777Z	INFO	leader	Not the leader. Waiting.
...

020-07-20T06:24:37.437Z	INFO	leader	Not the leader. Waiting.
```

```
kubectl get po -n operator -o wide
NAME                                                              READY   STATUS        RESTARTS   AGE    IP                NODE           NOMINATED NODE   READINESS GATES
helm-controller-5f89f89569-554ld                                  1/1     Running       0          135m   192.168.99.47     bluven-k8s-3   <none>           <none>
helm-controller-5f89f89569-mkksl                                  0/1     Terminating   0          162m   192.168.69.81     bluven-k8s-4   <none>           <none>
redis-operator-redisoperator-8f666f54f-vwwdl                      1/1     Running       0          19d    192.168.132.121   bluven-k8s-2   <none>           <none>
```

**Environment**
* operator-sdk version: 

operator-sdk version: "v0.17.0", commit: "2fd7019f856cdb6f6618e2c3c80d15c3c79d1b6c"

* go version:

go version go1.13.6 linux/amd64

* Kubernetes version information:
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-11T18:14:22Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:12:17Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
```
* Kubernetes cluster kind: 

kubeadm

* Are you writing your operator in ansible, helm, or go?
 go

**Comment by bluven** [comment]
I changed leader lock to controller-runtime leader lock which is lease based.

**Comment by camilamacedo86** [comment]
Hi @bluven,

The new layout will use the controller-runtime leader by default and will be available for 1.0.0.. So, it will be solved. Thank you for let us know and share your solution as well.
